sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -877,6 +878,74 @@ case class GpuFromUnixTime(
   override lazy val resolved: Boolean = childrenResolved && checkInputDataTypes().isSuccess
 }
 
+
+class FromUTCTimestampExprMeta(
+    expr: FromUTCTimestamp,
+    override val conf: RapidsConf,
+    override val parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+  extends BinaryExprMeta[FromUTCTimestamp](expr, conf, parent, rule) {
+
+  private var timezoneIsNull : Boolean = false
+
+  override def tagExprForGpu(): Unit = {
+    val timezoneShortID = extractStringLit(expr.right).getOrElse("")
+
+    if(timezoneShortID == null) {
+      timezoneIsNull = true
+    } else {
+      val utc = ZoneId.of("UTC").normalized
+      // This is copied from Spark, to convert `(+|-)h:mm` into `(+|-)0h:mm`.
+      val timezone = ZoneId.of(timezoneShortID.replaceFirst("(\\+|\\-)(\\d):", "$10$2:"),
+        ZoneId.SHORT_IDS).normalized
+
+      if (timezone != utc) {
+        willNotWorkOnGpu("only timezones equivalent to UTC are supported")
+      }
+    }
+  }
+
+  override def convertToGpu(timestamp: Expression, timezone: Expression): GpuExpression =
+    GpuFromUTCTimestamp(timestamp, timezone, timezoneIsNull)
+}
+
+case class GpuFromUTCTimestamp(timestamp: Expression,
+                               timezone: Expression,
+                               timezoneIsNull: Boolean)
+  extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  override def left: Expression = timestamp
+  override def right: Expression = timezone
+  override def inputTypes: Seq[AbstractDataType] = Seq(TimestampType, StringType)
+  override def dataType: DataType = TimestampType
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    throw new IllegalStateException(
+      "Cannot have time zone given by a column vector in GpuFromUTCTimestamp")
+  }
+
+  override def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector = {
+    throw new IllegalStateException(
+      "Cannot have time zone given by a column vector in GpuFromUTCTimestamp")
+  }
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
+    if(timezoneIsNull) {
```

Comment:
Do we really need `timezoneIsNull` as a parameter to this class?  Seems like we can check `!rhs.getBase.isValid` rather than premeditate the literal null.